dataprofiler/data_readers/graph_data.py

```diff
@@ -0,0 +1,114 @@
+        if options.get("source_list", None) is None:
+            options.update(source_list = ['source', 'src', 'origin'])
+        if options.get("destination_list", None) is None:
+            options.update(destination_list = ['target', 'destination', 'dst'])
+        if options.get("source_node", None) is None:
+            options.update(source_node = self._find_target_string_in_column(options.get("column_name", None), options.get("source_list", None)))
+        if options.get("destination_node", None) is None:
+            options.update(destination_node = self._find_target_string_in_column(options.get("column_name", None), options.get("destination_list", None)))
+
+        #return self._load_data()
+
+    @classmethod
+    def _find_target_string_in_column(self, column_names, keyword_list):
+        '''
+        Find whether one of the columns names contains a keyword that could refer to a target node column
+        '''
+
+        column_name_symbols = ['_', '.', '-']
+        has_target = False
+        target_index = -1
+        
+        # iterate through columns, keywords, and delimiter name symbols to see if any permutation is contained in column names
+        for column in range(0, len(column_names)):
+            for keyword in keyword_list:
+                for symbol in column_name_symbols:
+                    
+                    append_start_word = symbol + keyword
+                    append_end_word = keyword + symbol
+
+                    if append_start_word in column_names[column] or append_end_word in column_names[column]:
+                        target_index = column
+                        has_target = True
+                        break
+            if has_target:
+                break
+        
+        return target_index
+
+
+    @classmethod
+    def csv_column_names(cls, file_path, options):
+        '''
+        fetches a list of column names from the csv file
+        '''
+
+        column_names = []
+
+        with open(file_path) as csv_file:
+            csv_reader = csv.reader(csv_file, delimiter = options.get("delimiter", None))
+            
+            # fetch only column names
+            for row in csv_reader:
+                column_names.append(row)
+                break
+        
+        column_names = column_names[0]
+
+        # replace all whitespaces in the column names
+        for index in range(0, len(column_names)):
+            column_names[index] = column_names[index].replace(" ", "")
+
+        return column_names
+
+
+    @classmethod
+    def is_match(cls, file_path, options):
+        '''
+        Determines whether the file is a graph
+        Current formats checked:
+            - attributed edge list
+
+        This works by finding whether the file contains a target and a source node
+        '''
+
+        if options is None:
+            options = dict()
+        if CSVData.is_match(file_path, options):
```

Comment:
I may have misspoke before.
I think this should be:
`if not CSVData.is_match(file_path, options):`
^^ Are we not guaranteeing that it is a CSV to be read?